public/src/client/flags/list.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+/* globals define */
+
+define('forum/flags/list', ['components', 'Chart'], function (components, Chart) {
+	var Flags = {};
+
+	Flags.init = function () {
+		Flags.enableFilterForm();
+		Flags.enableChatButtons();
+		Flags.handleGraphs();
+	};
+
+	Flags.enableFilterForm = function () {
+		var filtersEl = components.get('flags/filters');
+
+		// Parse ajaxify data to set form values to reflect current filters
+		for (var filter in ajaxify.data.filters) {
+			filtersEl.find('[name="' + filter + '"]').val(ajaxify.data.filters[filter]);
+		}
+
+		filtersEl.find('button').on('click', function () {
+			var payload = filtersEl.serializeArray();
+			var qs = payload.map(function (filter) {
+				if (filter.value) {
+					return filter.name + '=' + filter.value;
+				} else {
+					return;
+				}
+			}).filter(Boolean).join('&');
```

Comment:
This can be replaced with 

```
var payload = filtersEl.serializeArray();
var qs = $.param(payload);
ajaxify.go('flags?' + qs);
```
http://api.jquery.com/jquery.param/